### PR TITLE
KEYCLOAK-5416 Migration from 3.2.1 to 3.3.0 doesn't work on MSSQL due to constraint violation

### PR DESCRIPTION
**Pull request by oguzkilcan**
… to constraint violation

### CODE REVIEW

File: model/jpa/src/main/resources/META-INF/jpa-changelog-3.2.0.xml [hunk 1]
```diff
@@ -29,7 +29,11 @@
             <changeSetExecuted id="3.2.0" author="keycloak" changeLogFile="META-INF/jpa-changelog-3.2.0.xml"/>
         </preConditions>
```
Comment (hmlnarik) [inline model/jpa/src/main/resources/META-INF/jpa-changelog-3.2.0.xml]: Id of this changeset (line 27) has to be changed to e.g. `3.2.0-fix-with-keycloak-5416`, and the following precondition needs to be added into `<preConditions>` tag:
```xml
<not>
    <changeSetExecuted id="3.2.0-fix" author="keycloak" changeLogFile="META-INF/jpa-changelog-3.2.0.xml"/>
</not>
```

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thanks for the fix. As it is now, it may fail on existing installations where application of 3.2.0-fix has already succeeded. Please adjust as per inline comment.

**Comment by hmlnarik** [comment]
@ouzklcn Thanks, that looks good. Could you please squash the changes into a single commit?

**Comment by hmlnarik** [comment]
I've checked the new commit, and on mysql the new changeset sadly fails:
```
[...]
Caused by: java.sql.SQLException: Cannot drop index 'IDX_CLIENT_INIT_ACC_REALM': needed in a foreign key constraint
```

So instead of changing existing changeset, what is needed here is to have changeset specific to mssql:
- leave the original changeset, only add a precondition `<not><dbms type="mssql"/></not>`
- copy it to another changeset with the changes from this PR and with precondition `<dbms type="mssql"/>`

**Comment by hmlnarik** [comment]
Thanks, I'm going to check it now. Hopefully all DBs would be happy now.

**Comment by hmlnarik** [comment]
All DBs seem happy with the current version.
@ouzklcn Thank you for the PR.
@stianst This PR can be merged.

**Comment by stianst** [comment]
Please send PR to 3.3.x branch as well.

**Comment by hmlnarik** [comment]
PR for 3.3.x: https://github.com/keycloak/keycloak/pull/4479

**Pull request merged**